cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield.hpp

```diff
@@ -160,7 +160,7 @@ template <typename Composer, typename T> class bigfield {
         field_t<Composer> lo = binary_basis_limbs[0].element + (binary_basis_limbs[1].element * shift_1);
         field_t<Composer> hi = binary_basis_limbs[2].element + (binary_basis_limbs[3].element * shift_1);
         // n.b. this only works if NUM_LIMB_BITS * 2 is divisible by 8
-        ASSERT((NUM_LIMB_BITS / 8) * 8 == NUM_LIMB_BITS);
+        ASSERT((NUM_LIMB_BITS * 2 / 8) * 8 == NUM_LIMB_BITS * 2);
```

Comment:
I see the comment above. Weird that this wasn't violated before. Was this assert triggering? Probably we should leave a comment here explaining the before-and-after so that someone can double check during a future audit when this logic is valid.

Reply:
Yes, this was triggering because the method `to_byte_array()` was being used in to convert an ECDSA public key to `byte_array`. I think we never really used the conversion of a bigfield scalar to `byte_array` (we still don't actually, the ECDSA conversion I mentioned is no longer required and has been removed). But I still kept this fix.

Apparently, Guido had bumped into this a while back in his fuzzing: https://github.com/AztecProtocol/aztec2-internal/issues/1496

I have also added a comment explaining the need for this assertion. I've removed your TODO.